src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp

```diff
@@ -1175,6 +1162,10 @@ void ShenandoahConcurrentGC::op_final_roots() {
       ShenandoahGenerationalHeap::heap()->update_region_ages(_generation->complete_marking_context());
     }
   }
+
+  if (VerifyAfterGC) {
```

Comment:
What are the conventions of when to use Verify{Before,After,During}GC on the one hand, vs ShenandoahVerify, G1Verify* etc., on the other?

Reply:
I don't really think there is a convention. In this particular case, it was "verifying" before concurrent reference processing was complete, which could lead to erroneous verification failures.